# Pandas Workbook

In [7]:
import numpy as numpy
import pandas as pd
import string

### Task 0
Write a function def from_numpy(array): that creates a pd.DataFrame from a np.ndarray:

* array is the np.ndarray from which you should create the pd.DataFrame
* The columns of the pd.DataFrame should be labeled in alphabetical order and capitalized. There will not be more than 26 columns.
* Returns: the newly created pd.DataFrame

In [8]:
def from_numpy(array):
    """
    Creates a pd.DataFrame from a np.ndarray:

    Inputs:
    array - np.ndarray

    Returns:
    pd.DataFrame
    """
    # Get numb er of columns
    num_columns = array.shape[1]

    # Create column names from alphabet
    columns = list(string.ascii_uppercase[:num_columns])

    # Create DF
    df = pd.DataFrame(array, columns=columns)

    return df

In [9]:
#!/usr/bin/env python3

import numpy as np
# from_numpy = __import__('0-from_numpy').from_numpy

np.random.seed(0)
A = np.random.randn(5, 8)
print(from_numpy(A))
B = np.random.randn(9, 3)
print(from_numpy(B))

          A         B         C         D         E         F         G  \
0  1.764052  0.400157  0.978738  2.240893  1.867558 -0.977278  0.950088   
1 -0.103219  0.410599  0.144044  1.454274  0.761038  0.121675  0.443863   
2  1.494079 -0.205158  0.313068 -0.854096 -2.552990  0.653619  0.864436   
3  2.269755 -1.454366  0.045759 -0.187184  1.532779  1.469359  0.154947   
4 -0.887786 -1.980796 -0.347912  0.156349  1.230291  1.202380 -0.387327   

          H  
0 -0.151357  
1  0.333674  
2 -0.742165  
3  0.378163  
4 -0.302303  
          A         B         C
0 -1.048553 -1.420018 -1.706270
1  1.950775 -0.509652 -0.438074
2 -1.252795  0.777490 -1.613898
3 -0.212740 -0.895467  0.386902
4 -0.510805 -1.180632 -0.028182
5  0.428332  0.066517  0.302472
6 -0.634322 -0.362741 -0.672460
7 -0.359553 -0.813146 -1.726283
8  0.177426 -0.401781 -1.630198


### Task 1

Write a python script that created a pd.DataFrame from a dictionary:

* The first column should be labeled First and have the values 0.0, 0.5, 1.0, and 1.5
* The second column should be labeled Second and have the values one, two, three, four
* The rows should be labeled A, B, C, and D, respectively
* The pd.DataFrame should be saved into the variable df